cardano-testnet/src/Testnet/Start/Cardano.hs

```diff
@@ -418,6 +418,9 @@ cardanoTestnet
 
     H.assertExpectedSposInLedgerState stakePoolsFp testnetOptions execConfig
 
+    when (cardanoEnableNewEpochStateLogging testnetOptions) $
```

Comment:
Can we write a test to confirm this gets killed?

Reply:
I've upstreamed concurrency abstractions to https://github.com/input-output-hk/hedgehog-extras/pull/66/files and added a test `TestWatchdogSpec` which tests that `asyncRegistered_` gets killed by watchdog. I'll add a test there checking if it's properly cleaned up by `MonadResource`.